src/app/settings/action-triggers/components/edit-action-triggers/edit-action-triggers.component.ts

```diff
@@ -0,0 +1,150 @@
+  }
+
+  private onActionsChanged(loadedActions: Tv2PartAction[]): void {
+    this.actions = loadedActions
+    this.loading = false
+  }
+
+  private patchValue(actionValue: ActionTrigger<KeyboardAndSelectionTriggerData>): void {
+    this.actionForm.patchValue({ actionId: actionValue.actionId, data: { keys: actionValue.data.keys.join(' + '), actionArguments: actionValue.data.actionArguments } })
+  }
+
+  public onKeyDown(event: KeyboardEvent): void {
+    event.preventDefault()
+    const newKeyCode: string = SHORTCUT_KEYS_MAPPINGS[event.code] ? SHORTCUT_KEYS_MAPPINGS[event.code] : event.code
+    if (this.keyPress) {
+      const currentKeys: string[] = this.actionForm?.get('data')?.get('keys')?.value ? this.actionForm?.get('data')?.get('keys')?.value.split(' + ') : []
+      if (currentKeys.findIndex(keyCode => keyCode === newKeyCode) === -1) {
+        currentKeys.push(newKeyCode)
+        this.actionForm?.get('data')?.get('keys')?.patchValue(currentKeys.join(' + '))
+      }
+    } else {
+      this.actionForm?.get('data')?.get('keys')?.patchValue(newKeyCode)
+      this.keyPress = true
+    }
+  }
+
+  public selectNewActionTrigger(action: Tv2PartAction): void {
+    this.actionForm.patchValue({ actionId: action.id })
+  }
+
+  public onKeyUp(): void {
+    this.keyPress = false
+  }
+
+  public get isUpdateAction(): boolean {
+    return !!this.selectedActionTrigger && !!this.selectedActionTrigger.id
+  }
+
+  public submit(): void {
+    this.submitted = true
+    if (this.actionForm.invalid) {
+      this.actionForm.markAllAsTouched()
+      return
+    }
+    this.submitting = true
+    const actionTriggerValue = this.prepareSendData(this.actionForm.value)
+    if (this.isUpdateAction) {
+      this.updateActionTrigger({ ...actionTriggerValue, id: this.selectedActionTrigger?.id as string })
+    } else {
+      this.createActionTrigger(actionTriggerValue)
+    }
+  }
+
+  private prepareSendData(formData: ActionTriggerUIForm): CreateActionTrigger<KeyboardTriggerData> {
+    const keysArray: string[] = formData.data?.keys?.split(' + ')
+    const resultData: CreateActionTrigger<KeyboardTriggerData> = { actionId: formData.actionId, data: { keys: keysArray, actionArguments: +formData.data.actionArguments } }
+    return resultData
+  }
+
+  private createActionTrigger(actionTrigger: CreateActionTrigger<KeyboardTriggerData>): void {
+    this.actionTriggerService.createActionTrigger(actionTrigger).subscribe({
+      next: () => {
+        this.actionForm.reset()
+      },
+    })
+  }
+
+  private updateActionTrigger(actionTrigger: ActionTrigger<KeyboardTriggerData>): void {
+    this.actionTriggerService.updateActionTrigger(actionTrigger).subscribe()
+  }
+
+  public cancelEdit(): void {
+    this.cancelActionTrigger.emit()
+  }
+
+  public control(name: string): AbstractControl {
+    return this.actionForm.controls[name]
+  }
+
+  public showErrorFor(name: string, groupName?: string): boolean | undefined {
```

Comment:
There is a mismatch between the name of the method and what the method does.
The name of the method indicates that you are showing errors. But in reality your are just checking the state of your formControls (checking if they are invalid and touched).
Checking the state of your formControls is not what is showing the errors. The errors are being shown by CSS that is being applied when this method returns true which is something that is outside the scope of the method.

This is a case where the method has been named after _where/how_ you want to use it, but not after _what_ it actually does.
A more accurate name would be `isFormControlInvalid` or `shouldFormControlShowError`.

I think what makes the mismatch is that you have used a verb (show) and verbs indicates that something happens i.e. if I have a method called `showWarning()` I expect that a warning will appear whenever this method is called. However, if I have prefixed it with `should` then I would expect that the methods just checks whether a warning should be displayed, but it won't actually display it. So I would have code that looked like this:
```
if (this.shouldShowWarning()) {
  this.showWarning()
}
```
What has happend for your method is that is has been named after the body of the above if-statement, when in reality it is the conditional of the if-statement.

Reply:
The name of the method is changed to `isFormControlInvalid ` and all checks all ready happened in script part not in to the html, because of the changes in to the UI.